BraveWallet/Settings/ManageSiteConnectionsView.swift

```diff
@@ -0,0 +1,216 @@
+                  Button(role: .destructive, action: {
+                    withAnimation {
+                      siteConnectionStore.removeAllPermissions(from: [siteConnection])
+                    }
+                  }) {
+                    Label(Strings.Wallet.delete, systemImage: "trash")
+                  }
+                }
+            } else {
+              content
+            }
+          }
+        }
+      }
+      .onDelete { indexes in
+        let visibleSiteConnections = siteConnectionStore.siteConnections.filter(by: filterText)
+        let siteConnectionsToRemove = indexes.map { visibleSiteConnections[$0] }
+        withAnimation {
+          siteConnectionStore.removeAllPermissions(from: siteConnectionsToRemove)
+        }
+      }
+    }
+    .listStyle(.insetGrouped)
+    .navigationTitle(Strings.Wallet.web3PreferencesManageSiteConnections)
+    .navigationBarTitleDisplayMode(.inline)
+    .filterable(text: $filterText, prompt: Strings.Wallet.manageSiteConnectionsFilterPlaceholder)
+    .toolbar {
+      ToolbarItemGroup(placement: .bottomBar) {
+        Spacer()
+        Button(action: {
+          isShowingConfirmAlert = true
+        }) {
+          Text(Strings.Wallet.manageSiteConnectionsRemoveAll)
+            .foregroundColor(siteConnectionStore.siteConnections.isEmpty ? Color(.braveDisabled) : .red)
+        }
+        .disabled(siteConnectionStore.siteConnections.isEmpty)
+      }
+    }
+    .onAppear(perform: siteConnectionStore.fetchSiteConnections)
+    .alert(isPresented: $isShowingConfirmAlert) {
+      Alert(
+        title: Text(Strings.Wallet.manageSiteConnectionsConfirmAlertTitle),
+        message: Text(Strings.Wallet.manageSiteConnectionsConfirmAlertMessage),
+        primaryButton: Alert.Button.destructive(
+          Text(Strings.Wallet.manageSiteConnectionsConfirmAlertRemove),
+          action: removeAll
+        ),
+        secondaryButton: Alert.Button.cancel(Text(Strings.CancelString))
+      )
+    }
+  }
+  
+  func removeAll() {
+    let visibleSiteConnections = siteConnectionStore.siteConnections.filter(by: filterText)
+    siteConnectionStore.removeAllPermissions(from: visibleSiteConnections)
+    filterText = ""
+  }
+}
+ 
+private struct SiteRow: View {
+
+  let siteConnection: SiteConnection
+
+  private let maxBlockies = 3
+  @ScaledMetric private var blockieSize: CGFloat = 16
+  private let maxBlockieSize: CGFloat = 32
+  @ScaledMetric private var blockieDotSize: CGFloat = 2
+  
+  var body: some View {
+    VStack(alignment: .leading, spacing: 6) {
+      Text(verbatim: siteConnection.url)
+        .foregroundColor(Color(.bravePrimary))
+      HStack {
+        Text(String.localizedStringWithFormat(Strings.Wallet.manageSiteConnectionsAccount, siteConnection.connectedAddresses.count, siteConnection.connectedAddresses.count == 1 ? Strings.Wallet.manageSiteConnectionsAccountSingular : Strings.Wallet.manageSiteConnectionsAccountPlural))
```

Comment:
that's a long line, maybe split it.
We try to not go over 120-140 line length if possible